introducing-pgstream.mdx

```diff
@@ -0,0 +1,96 @@
+---
+title: 'Introducing pgstream: Postgres replication with DDL changes'
+description: 'Today we’re excited to expand our open source Postgres platform with pgstream, a CDC command line tool and library for PostgreSQL with replication support for DDL changes to any provided output.'
+image:
+  src: /images/pgstream/introducing-pgstream@2x.jpg
+  alt: pgstream
+author: Esther Minano Sanz
+authorEmail: esther@xata.io
+date: 07-17-2024
+tags: ['open-source', 'postgres', 'fpReplication', 'fpPgstream', 'fpPgroll', 'fpPgzx', 'oss']
+published: false
+slug: pgstream-postgres-replication-schema-changes
+ogImage: /images/pgstream/introducing-pgstream@2x.jpg
+---
+
+## Why?
+
+At Xata, Postgres takes centre stage. And while it is our main database, we also offer other features that require us to extend its reach while keeping the data in sync. A good example of this is our [full-text search](https://xata.io/full-text-search) feature, which enables the use of Elasticsearch on top of Postgres. To keep these two datastores in sync, we capture and identify data and schema changes in Postgres and push these modifications downstream to Elasticsearch with minimal latency. This is often referred to as CDC (Change Data Capture).
+
+So now you know our use case, but why did we build our own replication tool? There's many established solutions out there, but we had very specific requirements, which included support for continuous tracking of schema changes (DDL). This was something that existing tooling didn't support at the time. Database schemas have a tendency to change over time - if your CDC tool doesn't support replicating them, you risk data loss and manual intervention to fix your pipeline. There had to be a better way!
+
+We also wanted a solution that was easy to deploy and operate for both big and small setups, which isn't always the case for existing tooling.
+
+## Introducing pgstream
+
+And so [pgstream](https://github.com/xataio/pgstream) was born! Pgstream is an open source CDC command-line tool and library. Some of its key features include:
+
+- **Schema change tracking and replication of DDL changes:** it's no surprise, since this was one of our biggest requirements, that it made it as an integral feature of pgstream. We will go into a bit more detail on how this is implemented below.
+- **Modular deployment configuration:** pgstream modular implementation allows it to be configured for simple use cases, removing unnecessary complexity and deployment challenges - the only hard requirement for pgstream is a Postgres database! However, it can also easily integrate with Kafka for more complex usecases.
```

Comment:
In this case I think using difficult requirement would change the meaning. I used hard requirement to highlight that postgres is the only thing required (must have) for pgstream, rather than something difficult to get.

Reply:
Sounds good!